integration-tests/integration-tests-quarkus-processes-persistence/integration-tests-processes-persistence-common/src/test/java/org/kie/kogito/it/PersistenceTest.java

```
@@ -225,4 +229,26 @@ void testMultipleInstance() {
                 .then()
                 .statusCode(404);
     }
+
+    @Test
+    void testAsyncWIH() {
+        String pId = given().contentType(ContentType.JSON)
+                .pathParam("processId", PROCESS_ASYNC_WIH)
+                .when()
+                .post("/{processId}")
+                .then()
+                .statusCode(201)
+                .body("id", not(emptyOrNullString()))
+                .extract()
+                .path("id");
+
+        await().atMost(TIMEOUT)
```

The course of test is here following?
1. Start process with custom workitem, get id.
2. Check that process with that id has finished

?

From the Jira I got the impression we would like to allow for a more complex behavior:
```
Marshallers are calling disconnect every time a process is persisted. Once we call this method, the current ProcessManager removes the persisted instance from memory.

If a Async work item handler is trying to use that manager to complete a work item when the future task is completed, it will fail because it want be able to find the process instance, something that did not happen when persistence is not active
```

So the original reported issue would most probably fail silently in this test wouldn't it? Cause it's the async wih that fails - will that result in any relevant outcome in the course of this test case? I don't see it.

---

@jstastny-cz the process would simply not finish if there is any issue while executing the WIH. As the WIH execution is async, the process state is persisted on the node where the WIH will be later executed. If it fails to execute, the process will simply get stuck there and not complete.

---

Thank you for explaining.